Sources/eppo/EppoClient.swift

```diff
@@ -55,40 +57,71 @@ public class EppoClient {
         subjectAttributes: SubjectAttributes = SubjectAttributes(),
         defaultValue: Bool) throws -> Bool
     {
-        return try getInternalAssignment(
-            flagKey: flagKey, 
-            subjectKey: subjectKey, 
-            subjectAttributes: subjectAttributes, 
-            useTypedVariationValue: true
-        )?.boolValue() ?? defaultValue
+        do {
+            return try getInternalAssignment(
+                flagKey: flagKey, 
+                subjectKey: subjectKey, 
+                subjectAttributes: subjectAttributes,
+                expectedVariationType: UFC_VariationType.boolean
+            )?.variation?.value.getBoolValue() ?? defaultValue
+        } catch {
+            // todo: implement graceful mode
+            return defaultValue
+        }
     }
     
-    public func getJSONStringAssignment(
+    // todo: add back when supporting JSON
+//    public func getJSONAssignment(
+//        flagKey: String,
+//        subjectKey: String,
+//        subjectAttributes: SubjectAttributes,
+//        defaultValue: [String: EppoValue]) throws -> [String: EppoValue]
+//    {
+//        return try getInternalAssignment(
+//            flagKey: flagKey, 
+//            subjectKey: subjectKey, 
+//            subjectAttributes: subjectAttributes
+//        )?.variation?.value?.objectValue() ?? defaultValue
+//    }
+//    
+    
+    public func getIntegerAssignment(
         flagKey: String,
         subjectKey: String,
-        subjectAttributes: SubjectAttributes,
-        defaultValue: String) throws -> String
+        subjectAttributes: SubjectAttributes = SubjectAttributes(),
+        defaultValue: Int) throws -> Int
```

Comment:
No expert on Swift, so just double checking -- this function should not throw errors right?

Reply:
It does throw so the caller has to handle that; the code won't compile for them with a handle.

We added the safe mode to other SDKs because those don't have the throwable trait; Java has this as well. So developers won't be surprised by unexpected exceptions.

Reply:
Plenty of them - just to name a few if a `flagKey` or `subjectKey` is empty and others. I haven't yet added explicit support for graceful mode in swift but I am thinking it is not as necessary as in other SDKs since the problem graceful mode was trying to solve was when users were surprised by an exception. By having a `throws -> Int` type they will now to handle the return from assignment methods.

We can't have it both ways though: if we wrap each public method in a `do/catch` block and want to remove `throws` then the `catch` has to mask all exceptions. Conditionally throwing is still throwing.

Is there something I am missing or a concern that exceptions will leak unexpectedly to users?

Reply:
I ask because the body of the method is wrapped in a do/catch, so my expectation would be that all exceptions are handled within, and that callers of this method have no exceptions to handle themselves

Reply:
That's fair - what I meant was I want those exceptions to be returned and have a follow up (https://linear.app/eppo/issue/FF-1953/add-graceful-mode-to-ios-sdk) to address it. 

We'll need to allow exceptions to be thrown if we want to use that as a signaling method when something went wrong, or opt for log messages and always return the default message (current behavior).

Reply:
Gotcha, that plan makes sense to me, thanks for clarifying!